webtau-core/src/main/java/com/twosigma/webtau/Matchers.java

```diff
@@ -0,0 +1,268 @@
+package com.twosigma.webtau;
+
+import com.twosigma.webtau.expectation.*;
+import com.twosigma.webtau.expectation.code.ThrowExceptionMatcher;
+import com.twosigma.webtau.expectation.contain.ContainMatcher;
+import com.twosigma.webtau.expectation.equality.*;
+
+import java.util.regex.Pattern;
+
+/**
+ * Convenient place to discover all the available matchers
+ */
+public class Matchers {
+    /**
+     * Starting point of a value matcher
+     * <pre>
+     * actual(value).should(beGreaterThan(10));
+     * actual(value).shouldNot(beGreaterThan(10));
+     * </pre>
+     * Note: In Groovy you can just do <code>value.should beGreaterThan(10)</code>
+     * @param actual value to assert against
+     * @return Object to chain a matcher against
+     */
+    public static ActualValueExpectations actual(Object actual) {
+        return new ActualValue(actual);
+    }
+
+    /**
+     * Starting point of a code matcher
+     * <pre>
+     * code(() -> {
+     *    businessLogic(-10);
+     * }).should(throwException(IllegalArgumentException.class, "negative are not allowed"));
+     * </pre>
+     *
+     * @param codeBlock code to match against
+     * @return Object to chain a matcher against
+     */
+    public static ActualCodeExpectations code(CodeBlock codeBlock) {
+        return new ActualCode(codeBlock);
+    }
+
+    /**
+     * Equal matcher
+     * <pre>
+     * actual(value).should(equal(10));
+     * </pre>
+     * @param expected value to be equal to
+     * @return matcher instance
+     */
+    public static EqualMatcher equal(Object expected) {
+        return new EqualMatcher(expected);
+    }
+
+    /**
+     * Not equal matcher
+     * <pre>
+     * actual(value).should(notEqual(10));
+     * </pre>
+     * @param expected value to not be equal to
+     * @return matcher instance
+     */
+    public static NotEqualMatcher notEqual(Object expected) {
+        return new NotEqualMatcher(expected);
+    }
+
+    /**
+     * Contain matcher
+     * <pre>
+     * actual(value).should(contain(10));
```

Comment:
How about `actual(containValue).should(contain(10));` just to make it clearer that this is a matcher for collections.

Reply:
not sure what `containValue` adds here. I do agree that we need to show that it works for containers. `contain` works for strings to.

maybe something like this?
```
actual(collection).should(contain(10)); 
actual(text).should(contain("hello")); 
```

Reply:
Typo, brain to finger disconnect.

I meant `collectionValue` not `containValue`.